gnat2goto/driver/tree_walk.adb

```diff
@@ -3079,7 +3079,8 @@ package body Tree_Walk is
       function Needs_Default_Initialisation (E : Entity_Id) return Boolean is
       begin
          return Has_Defaulted_Discriminants (E)
-           or else Has_Defaulted_Components (E);
+           or else Has_Defaulted_Components (E)
```

Comment:
Should we consider a test for this?  (It previously got missed in the existing tests)

Reply:
Well I guess the new `arrays_index` test would fail if we only implemented `array_literal` but didn't enforce default initialisation here.